newlib/libc/picolib/meson.build

```diff
@@ -32,37 +32,19 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
 # OF THE POSSIBILITY OF SUCH DAMAGE.
 #
-srcs_machine_tls = {
-			'riscv' : ['machine/riscv/tls.c'],
-			'riscv32' : ['machine/riscv/tls.c'],
-			'riscv64' : ['machine/riscv/tls.c'],
-			'arm' : ['machine/arm/tls.c', 'machine/arm/read_tp.S'],
-			'aarch64' : [ 'machine/aarch64/tls.c' ]
-		      }
```

Comment:
We might want to keep this pattern.  we have a policy not to rename because of newlib. and newlib already uses `i386` for x86 and, well, it's inconsistent for amd64.  Anyway, we have to have some kind of mapping table for arch to directory structure anyway.